tests/visualization_tests/matplotlib_tests/test_optimization_history.py

```diff
@@ -96,12 +102,21 @@ def objective(trial: Trial) -> float:
     assert len(figure.get_legend().get_texts()) == n_studies
 
     # Test customized target name.
-    figure = plot_optimization_history(studies, target_name="Target Name")
-    assert (
-        figure.legend().get_texts()[n_studies].get_text()
```

Comment:
What `figure.legend().get_texts()[n_studies]` returns is also potentially undefined.